### ISSUE TITLE
GraphQL query breaks seemingly randomly

### ISSUE THREAD

**Issue by Paddy888** [gatsbyjs/gatsby#14388]
## Description

Sometimes after changing an image, Gatsby will fail to compile. Previously working, and no changes other than editing or deleting an image file. Sometimes deleting the cache and public folders will help fix it, but other times even rolling back numerous commits, it will still fail to compile

### Steps to reproduce

As noted above, it seems to mainly happen after editing or deleting an image file from the /static/img folder

### Expected result

gatsby develop should compile without error

### Actual result

GraphQL Error Field "image" must not have a selection since type "String" has no subfields.

file: /src/templates/product-page.js

 34 |             image {
     |                   ^
  35 |               childImageSharp {
  36 |                 fluid(maxWidth: 526, quality: 92) {
  37 |                   ...GatsbyImageSharpFluid
  38 |                 }

### Environment

 System:
    OS: Windows 10
    CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
  Binaries:
    npm: 6.2.0 - 
  Browsers:
    Edge: 42.17134.1.0
  npmPackages:
    gatsby: ^2.7.1 => 2.7.1
    gatsby-image: ^2.0.23 => 2.0.30
    gatsby-plugin-cookiehub: ^1.0.0 => 1.0.0
    gatsby-plugin-facebook-pixel: ^1.0.3 => 1.0.3
    gatsby-plugin-google-analytics: ^2.0.20 => 2.0.20
    gatsby-plugin-netlify: ^2.0.6 => 2.0.11
    gatsby-plugin-netlify-cms: ^3.0.9 => 3.0.12
    gatsby-plugin-purgecss: ^3.1.0 => 3.1.0
    gatsby-plugin-react-helmet: ^3.0.4 => 3.0.7
    gatsby-plugin-sass: ^2.0.7 => 2.0.10
    gatsby-plugin-sharp: ^2.1.1 => 2.1.1
    gatsby-remark-copy-linked-files: ^2.0.7 => 2.0.9
    gatsby-remark-embed-video: ^1.7.1 => 1.7.1
    gatsby-remark-images: ^3.0.1 => 3.0.5
    gatsby-remark-relative-images: ^0.2.2 => 0.2.2
    gatsby-source-filesystem: ^2.0.12 => 2.0.23
    gatsby-transformer-remark: ^2.1.15 => 2.2.6
    gatsby-transformer-sharp: ^2.1.9 => 2.1.14

**Comment by wardpeet** [comment]
Hi @Paddy888!

Sorry to hear you're running into an issue. To help us best begin debugging the underlying cause, it is incredibly helpful if you're able to create a minimal reproduction. This is a simplified example of the issue that makes it clear and obvious what the issue is and how we can begin to debug it.

If you're up for it, we'd very much appreciate if you could provide a minimal reproduction and we'll be able to take another look.

Thanks for using Gatsby! 💜

**Comment by Paddy888** [comment]
Not sure what you mean. 

I just rolled back a number of commits until I could get a working compile again. 

I began redoing the changes I had made one by one, stopping the dev server and running gatsby clean each time. 

I removed a number of image files from /static/img again cleaning and restarting the server each time. 

Eventually, I removed an image, which resulted in this error. I searched the project and do not find any instance of it being used in any of my files. Discarded the change, and was able to compile again.
